### PR TITLE
Use attn_mask_type of causal_padding for cudnn_flash_attention

### PR DESCRIPTION
**Pull request by bvandermoon**
### Notes

Update attn_mask_type so that it is not ignored. Related to issue [878](https://github.com/AI-Hypercomputer/maxtext/issues/878).

Note that this change currently only impacts workloads using `cudnn_flash_te` attention.

### Testing

* Trained 10 steps on GPUs with `cudnn_flash_te` enabled. Saw the same high-level output (step time, loss, etc.) before and after this change
* Ran the logit checker with llama2-7b logits and the test succeeded:

Screenshot 2024-10-01 at 2 06 07 PM

### CODE REVIEW

**Comment by gobbleturk** [comment]
> ### Notes
> Update attn_mask_type so that it is not ignored. Related to issue [878](https://github.com/AI-Hypercomputer/maxtext/issues/878).
> 
> Note that this change currently only impacts workloads using `cudnn_flash_te` attention.
> 
> ### Testing
> Trained 10 steps on GPUs with `cudnn_flash_te` enabled. Saw the same high-level output (step time, loss, etc.) before and after this change.
> 
> Note: The loss was suspicious since it dropped from 10.307 to 0 after the third step. But this happened before and after this change so it appears to be caused by something else

I assume the loss is dropping so quickly because you are using synthetic data (`dataset_type=synthetic`), you can use real data to measure the loss. Ideally we would test correctness with something like our [golden logits test](https://github.com/AI-Hypercomputer/maxtext/blob/5ae1b711bc94bd0003d1d0ad76a220f3a76fbe3b/MaxText/tests/forward_pass_logit_checker.py#L97)

**Review by gobbleturk — APPROVED** [review]
Confirmed golden test results offline!

**Comment by bvandermoon** [comment]
> > ### Notes
> > Update attn_mask_type so that it is not ignored. Related to issue [878](https://github.com/AI-Hypercomputer/maxtext/issues/878).
> > Note that this change currently only impacts workloads using `cudnn_flash_te` attention.
> > ### Testing
> > Trained 10 steps on GPUs with `cudnn_flash_te` enabled. Saw the same high-level output (step time, loss, etc.) before and after this change.
> > Note: The loss was suspicious since it dropped from 10.307 to 0 after the third step. But this happened before and after this change so it appears to be caused by something else
> 
> I assume the loss is dropping so quickly because you are using synthetic data (`dataset_type=synthetic`), you can use real data to measure the loss. Ideally we would test correctness with something like our [golden logits test](https://github.com/AI-Hypercomputer/maxtext/blob/5ae1b711bc94bd0003d1d0ad76a220f3a76fbe3b/MaxText/tests/forward_pass_logit_checker.py#L97)

Thanks Matt. That was the issue with the loss. Also ran the golden logits test and it passed. Updated the description with the test
